pkg/odo/genericclioptions/runnable.go

```diff
@@ -93,6 +93,11 @@ func GenericRun(o Runnable, cmd *cobra.Command, args []string) {
 		klog.V(4).Infof("WARNING: debug telemetry, if enabled, will be logged in %s", debugTelemetry)
 	}
 
+	err = scontext.SetCaller(cmd.Context(), scontext.CallerType(os.Getenv(segment.TelemetryCaller)))
+	if err != nil {
+		log.Warningf("Error handling caller property for telemetry: %v", err)
```

Comment:
Are we sure we want to print this as a user facing warning? Shouldn't we hide it to log level 3 or 4?

Reply:
It made sense to me when invalid values were not inserted to telemetry properties. Now that even invalid values are being sent out, yes, I think we can remove this warning.
I'll change it.

Reply:
Changes pushed - please take another look.